### DISCUSSION TITLE
error: call to undeclared function 'rmdir'

### DISCUSSION THREAD

**Discussion by mgregg128** [iraf-community/iraf#297] [Q&A]
Hello,
I am trying to install iraf-2.17 on my old 16" MacBook pro (intel) from 2017, running Ventura 13.3.  All goes fine with the download and install.  An error occurs at the make stage, generating the multiline error message:

```
sysdep.c:112:3: error: call to undeclared function 'rmdir'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
                rmdir(s);
                ^
sysdep.c:112:3: note: did you mean 'rmtdir'?
sysdep.c:107:1: note: 'rmtdir' declared here
rmtdir(Void)
^
1 error generated.
make[1]: *** [sysdep.o] Error 1
make: *** [sysgen] Error 2
```

At which point the build stops.  Any advice?  There are quite a few warning messages along the way as well.  
I have specified macintel as the architecture, and have the most recent Xcode installed.
I've attached the build.log file here.   

Thanks for any help or suggestions,

Michael

[build.log](https://github.com/iraf-community/iraf/files/11427709/build.log)

**Comment by olebole** [comment]
The latest XCode version for macOS is more picky than before, so you need an updated IRAF tarball. Unfortunately, there is no new version yet. You can however try out the current development snapshot, which can be downloaded [here](https://github.com/iraf-community/iraf/archive/refs/heads/main.tar.gz).
Note that the installation procedure has changed; the actual one can be found in the tarball as `INSTALL.md` or [here](https://github.com/iraf-community/iraf/blob/main/INSTALL.md).
Please report back whether this helped to solve your problem.

**Reply by mmorage** [reply]
I also have the same problem  I will try it, and I will let you know.

**Reply by olebole** [reply]
Just one thing: I originally erroneously linked in my bottom post to the `iraf-main.zip` file, which does not work. This is now corrected to the `iraf-main.tar.gz`.

**Reply by mmorage** [reply]
Compiled without issues and make test gives the expected result.  THANKS!!!

**Reply by mmorage** [reply]
I will try the tar.gz

**Reply by mmorage** [reply]
It compiled without problems. M2 chip

**Reply by mgregg128** [reply]
Hello,
Thank you, yes, this make and install appears to have worked for my system (macintel 64bit) too.  Although, I still have some linger environment variable assignments related to iraf that I need to fix.  Except for those, the install passes its own tests, and it does start with the ecl command and looks fine.  I'm including the build.log again, just in case it is of any interest.  
[build.log](https://github.com/iraf-community/iraf/files/11436643/build.log)
One other problem I have is that xgterm (X11iraf) refuses to run or build on my system.  I'll start a new thread for that... 

Thank you very much!

**Reply by mgregg128** [reply]
OK, perhaps spoke a little too soon.  While the compile and install works, the command mkiraf appears to fail.  It generates the familiar message: 
```
-- creating a new uparm directory
A new LOGIN.CL file has been created in the current directory.
You may wish to review and edit this file to change the defaults.
```
except that no LOGIN.CL (or login.cl) file appears in the current directory (or any other directory, as far as I can tell).  Iraf will start with the default system login.cl file, or with my old login.cl file, after I comment out all the stsdas package reference calls that used to be available on startup when I had stsdas installed.  So this is perhaps a non-fatal error, but I am wondering why there appears to be a problem with the login.cl file, and whether this is symptomatic of other problems.  Advice appreciated, and thank you again.

**Reply by olebole** [reply]
The message is incorrect here (and should be removed); **mkiraf** does not create a `login.cl` by default anymore, because in most cases this is not needed. If you need a local copy, use the **-c** flag:
```
mkiraf -c
```
